From 15e251ee6c53a195b8de0e699ae9dd9bf2fb6fa2 Mon Sep 17 00:00:00 2001
From: ArenM <aren@peacevolution.org>
Date: Sat, 18 Feb 2023 15:01:06 -0500
Subject: [PATCH sxmo-utils 5/9] Separate services that are provided by
 external packages

These services are provided by the packages for these things on distros
that use systemd, so if the packager replaces superd with systemd they
must be removed.
---
 Makefile                                          | 15 ++++++++++++---
 .../services => external-services}/dunst.service  |  0
 .../services => external-services}/mako.service   |  0
 .../mmsd-tng.service                              |  0
 .../pipewire-pulse.service                        |  0
 .../pipewire.service                              |  0
 .../pulseaudio.service                            |  0
 .../services => external-services}/vvmd.service   |  0
 .../wireplumber.service                           |  0
 .../services/autocutsel-primary.service           |  0
 configs/{superd => }/services/autocutsel.service  |  0
 configs/{superd => }/services/bonsaid.service     |  0
 .../{superd => }/services/sxmo-x11-status.service |  0
 .../services/sxmo_autosuspend.service             |  0
 .../services/sxmo_battery_monitor.service         |  0
 configs/{superd => }/services/sxmo_conky.service  |  0
 .../services/sxmo_desktop_widget.service          |  0
 .../{superd => }/services/sxmo_hook_lisgd.service |  0
 .../services/sxmo_menumode_toggler.service        |  0
 .../services/sxmo_modemmonitor.service            |  0
 .../services/sxmo_networkmonitor.service          |  0
 .../services/sxmo_notificationmonitor.service     |  0
 .../services/sxmo_soundmonitor.service            |  0
 configs/{superd => }/services/sxmo_wob.service    |  0
 .../services/unclutter-xfixes.service             |  0
 configs/{superd => }/services/unclutter.service   |  0
 26 files changed, 12 insertions(+), 3 deletions(-)
 rename configs/{superd/services => external-services}/dunst.service (100%)
 rename configs/{superd/services => external-services}/mako.service (100%)
 rename configs/{superd/services => external-services}/mmsd-tng.service (100%)
 rename configs/{superd/services => external-services}/pipewire-pulse.service (100%)
 rename configs/{superd/services => external-services}/pipewire.service (100%)
 rename configs/{superd/services => external-services}/pulseaudio.service (100%)
 rename configs/{superd/services => external-services}/vvmd.service (100%)
 rename configs/{superd/services => external-services}/wireplumber.service (100%)
 rename configs/{superd => }/services/autocutsel-primary.service (100%)
 rename configs/{superd => }/services/autocutsel.service (100%)
 rename configs/{superd => }/services/bonsaid.service (100%)
 rename configs/{superd => }/services/sxmo-x11-status.service (100%)
 rename configs/{superd => }/services/sxmo_autosuspend.service (100%)
 rename configs/{superd => }/services/sxmo_battery_monitor.service (100%)
 rename configs/{superd => }/services/sxmo_conky.service (100%)
 rename configs/{superd => }/services/sxmo_desktop_widget.service (100%)
 rename configs/{superd => }/services/sxmo_hook_lisgd.service (100%)
 rename configs/{superd => }/services/sxmo_menumode_toggler.service (100%)
 rename configs/{superd => }/services/sxmo_modemmonitor.service (100%)
 rename configs/{superd => }/services/sxmo_networkmonitor.service (100%)
 rename configs/{superd => }/services/sxmo_notificationmonitor.service (100%)
 rename configs/{superd => }/services/sxmo_soundmonitor.service (100%)
 rename configs/{superd => }/services/sxmo_wob.service (100%)
 rename configs/{superd => }/services/unclutter-xfixes.service (100%)
 rename configs/{superd => }/services/unclutter.service (100%)

diff --git a/Makefile b/Makefile
index 0eba191..9c730fb 100644
--- a/Makefile
+++ b/Makefile
@@ -2,6 +2,13 @@ DESTDIR=
 PREFIX:=/usr
 SHAREDIR=$(PREFIX)/share
 MANDIR=$(SHAREDIR)/man
+
+# use $(PREFIX)/lib/systemd/user for systemd integration
+SERVICEDIR:=$(PREFIX)/share/superd/services
+
+# Install services for packages outside sxmo
+EXTERNAL_SERVICES:=1
+
 SCDOC=scdoc
 
 .PHONY: install shellcheck
@@ -93,9 +100,11 @@ install-scripts: $(PROGRAMS)
 	mkdir -p "$(DESTDIR)$(PREFIX)/share/sxmo/appscripts"
 	cd scripts/appscripts && find . -name 'sxmo_*.sh' | xargs -I{} ln -fs "$(PREFIX)/bin/{}" "$(DESTDIR)$(PREFIX)/share/sxmo/appscripts/{}" && cd ../..
 
-
-	mkdir -p "$(DESTDIR)$(PREFIX)/share/superd/services"
-	install -m 0644 -t $(DESTDIR)$(PREFIX)/share/superd/services configs/superd/services/*
+	mkdir -p "$(DESTDIR)$(SERVICEDIR)"
+	install -m 0644 -t "$(DESTDIR)$(SERVICEDIR)" configs/services/*
+	if [ "$(EXTERNAL_SERVICES)" = "1" ]; then \
+		install -m 0644 -t "$(DESTDIR)$(SERVICEDIR)" configs/external-services/*; \
+	fi
 
 	@echo "-------------------------------------------------------------------">&2
 	@echo "NOTICE 1: Do not forget to add sxmo-setpermissions to your init system, e.g. for openrc: rc-update add sxmo-setpermissions default && rc-service sxmo-setpermissions start" >&2
diff --git a/configs/superd/services/dunst.service b/configs/external-services/dunst.service
similarity index 100%
rename from configs/superd/services/dunst.service
rename to configs/external-services/dunst.service
diff --git a/configs/superd/services/mako.service b/configs/external-services/mako.service
similarity index 100%
rename from configs/superd/services/mako.service
rename to configs/external-services/mako.service
diff --git a/configs/superd/services/mmsd-tng.service b/configs/external-services/mmsd-tng.service
similarity index 100%
rename from configs/superd/services/mmsd-tng.service
rename to configs/external-services/mmsd-tng.service
diff --git a/configs/superd/services/pipewire-pulse.service b/configs/external-services/pipewire-pulse.service
similarity index 100%
rename from configs/superd/services/pipewire-pulse.service
rename to configs/external-services/pipewire-pulse.service
diff --git a/configs/superd/services/pipewire.service b/configs/external-services/pipewire.service
similarity index 100%
rename from configs/superd/services/pipewire.service
rename to configs/external-services/pipewire.service
diff --git a/configs/superd/services/pulseaudio.service b/configs/external-services/pulseaudio.service
similarity index 100%
rename from configs/superd/services/pulseaudio.service
rename to configs/external-services/pulseaudio.service
diff --git a/configs/superd/services/vvmd.service b/configs/external-services/vvmd.service
similarity index 100%
rename from configs/superd/services/vvmd.service
rename to configs/external-services/vvmd.service
diff --git a/configs/superd/services/wireplumber.service b/configs/external-services/wireplumber.service
similarity index 100%
rename from configs/superd/services/wireplumber.service
rename to configs/external-services/wireplumber.service
diff --git a/configs/superd/services/autocutsel-primary.service b/configs/services/autocutsel-primary.service
similarity index 100%
rename from configs/superd/services/autocutsel-primary.service
rename to configs/services/autocutsel-primary.service
diff --git a/configs/superd/services/autocutsel.service b/configs/services/autocutsel.service
similarity index 100%
rename from configs/superd/services/autocutsel.service
rename to configs/services/autocutsel.service
diff --git a/configs/superd/services/bonsaid.service b/configs/services/bonsaid.service
similarity index 100%
rename from configs/superd/services/bonsaid.service
rename to configs/services/bonsaid.service
diff --git a/configs/superd/services/sxmo-x11-status.service b/configs/services/sxmo-x11-status.service
similarity index 100%
rename from configs/superd/services/sxmo-x11-status.service
rename to configs/services/sxmo-x11-status.service
diff --git a/configs/superd/services/sxmo_autosuspend.service b/configs/services/sxmo_autosuspend.service
similarity index 100%
rename from configs/superd/services/sxmo_autosuspend.service
rename to configs/services/sxmo_autosuspend.service
diff --git a/configs/superd/services/sxmo_battery_monitor.service b/configs/services/sxmo_battery_monitor.service
similarity index 100%
rename from configs/superd/services/sxmo_battery_monitor.service
rename to configs/services/sxmo_battery_monitor.service
diff --git a/configs/superd/services/sxmo_conky.service b/configs/services/sxmo_conky.service
similarity index 100%
rename from configs/superd/services/sxmo_conky.service
rename to configs/services/sxmo_conky.service
diff --git a/configs/superd/services/sxmo_desktop_widget.service b/configs/services/sxmo_desktop_widget.service
similarity index 100%
rename from configs/superd/services/sxmo_desktop_widget.service
rename to configs/services/sxmo_desktop_widget.service
diff --git a/configs/superd/services/sxmo_hook_lisgd.service b/configs/services/sxmo_hook_lisgd.service
similarity index 100%
rename from configs/superd/services/sxmo_hook_lisgd.service
rename to configs/services/sxmo_hook_lisgd.service
diff --git a/configs/superd/services/sxmo_menumode_toggler.service b/configs/services/sxmo_menumode_toggler.service
similarity index 100%
rename from configs/superd/services/sxmo_menumode_toggler.service
rename to configs/services/sxmo_menumode_toggler.service
diff --git a/configs/superd/services/sxmo_modemmonitor.service b/configs/services/sxmo_modemmonitor.service
similarity index 100%
rename from configs/superd/services/sxmo_modemmonitor.service
rename to configs/services/sxmo_modemmonitor.service
diff --git a/configs/superd/services/sxmo_networkmonitor.service b/configs/services/sxmo_networkmonitor.service
similarity index 100%
rename from configs/superd/services/sxmo_networkmonitor.service
rename to configs/services/sxmo_networkmonitor.service
diff --git a/configs/superd/services/sxmo_notificationmonitor.service b/configs/services/sxmo_notificationmonitor.service
similarity index 100%
rename from configs/superd/services/sxmo_notificationmonitor.service
rename to configs/services/sxmo_notificationmonitor.service
diff --git a/configs/superd/services/sxmo_soundmonitor.service b/configs/services/sxmo_soundmonitor.service
similarity index 100%
rename from configs/superd/services/sxmo_soundmonitor.service
rename to configs/services/sxmo_soundmonitor.service
diff --git a/configs/superd/services/sxmo_wob.service b/configs/services/sxmo_wob.service
similarity index 100%
rename from configs/superd/services/sxmo_wob.service
rename to configs/services/sxmo_wob.service
diff --git a/configs/superd/services/unclutter-xfixes.service b/configs/services/unclutter-xfixes.service
similarity index 100%
rename from configs/superd/services/unclutter-xfixes.service
rename to configs/services/unclutter-xfixes.service
diff --git a/configs/superd/services/unclutter.service b/configs/services/unclutter.service
similarity index 100%
rename from configs/superd/services/unclutter.service
rename to configs/services/unclutter.service
-- 
2.39.2

